### ISSUE TITLE
Siamese Resnet50 problem with BN

### ISSUE THREAD

**Issue by owang** [fchollet/keras#4808]
Is there any way to set up a resnet50 in a siamese fashion?  E.g.,
```
input1 = Input(shape=(imh, imw, inputChannels))
input2 = Input(shape=(imh, imw, inputChannels))
model = ResNet50(weights='imagenet', include_top=True)
embed1 = model(input1)
embed2 = model(input2)
distance = Lambda(euclidean_distance)([embed1, embed2])
model = Model(input=[input1, input2], output=distance)
```

Does not work because:

> Exception: You are attempting to share a same `BatchNormalization` layer across different data flows. This is not possible. You should use `mode=2` in `BatchNormalization`. 

But if I change the resnet model then I would need to retrain on imagenet right? :| Is there a workaround? Thanks!

**Comment by JWGS1** [comment]
I am having the same issue, did you find a solution? I want to fine tune pretrained resnet50 in a Siamese network. Thanks.

**Comment by fchollet** [comment]
The solution is to update Keras to the current version at `master`. This is
a very old limitation.

**Comment by longchuan1985** [comment]
I have encountered the same issue, and got it resolved after updated into tensorflow-gpu 1.4.1 together with cudnn.so.6
